### PR TITLE
Feat: Generate and Claim Gyms rewards

### PR DESCRIPTION
**Pull request by PedroChaparro**
## Includes: 

- [x] Insert loomballs in the database setup. 
- [x] Add gym_rewards fields in the gyms collection.  
- [x] Add a probability to each object and loomball to become a gym reward for common users. 
- [x] Add a probability to each object and loomball to become a gym reward for gyms owners. 
- [x] Script to remove current rewards and generate new ones. (This script should be runned as a cronjob). 
- [x] Endpoint to claim the current rewards of some gym.  

## How can I test it?

1. Drop all collections from the mongo database (Except the users one). This is not mandatory but it will prevent duplicated and outdated entries. 
2. Go to the `algorithms/database` folder and re-generate the database.
3. From the same folder, run the `update:rewards` command, that will generate initial rewards for all the gyms. 
4. Send a POST request to the new `/gyms/claim_reward` endpoint, it should: 

- Add the user id to the `rewards_claimed_by` array in the claimed gym. 
image

- Add the items to the `inventory` field in the users collection.
image

### CODE REVIEW

**Comment by PedroChaparro** [comment]
@andres123dbh if this pull request goes merged, please update your structs in the `_getItemsUser` branch.

**Review by SilviaPabon — APPROVED** [review]
image
It works! Great!

File: algorithms/database/models/mongoose.js [hunk 1]
```diff
@@ -24,6 +24,42 @@ const GymSchema = new Schema(
...
+          reward_id: {
+            type: Schema.Types.ObjectId,
+            // Dynamically reference the correct collection
+            refPath: "current_rewards.reward_type",
+          },
+          reward_quantity: Number,
+        },
+      ],
+    },
+    current_owners_rewards: {
```
Comment (Woynert) [inline algorithms/database/models/mongoose.js:45]: Property `current_owners_rewards` defined here.

File: api/interfaces/interfaces.go [hunk 2]
```diff
@@ -61,22 +61,59 @@ type ZoneWithPopulatedLoomies struct {
...
-	Latitude  float64            `json:"latitude"      bson:"latitude"`
-	Longitude float64            `json:"longitude"      bson:"longitude"`
-	Name      string             `json:"name"      bson:"name"`
+	Id                    primitive.ObjectID   `json:"_id" bson:"_id"`
+	Latitude              float64              `json:"latitude"      bson:"latitude"`
+	Longitude             float64              `json:"longitude"      bson:"longitude"`
+	Name                  string               `json:"name"      bson:"name"`
+	Owner                 primitive.ObjectID   `json:"owner,omitempty"      bson:"owner,omitempty"`
+	CurrentPlayersRewards []GymRewardItem      `json:"current_players_rewards"      bson:"current_players_rewards"`
+	CurrentOwnerRewards   []GymRewardItem      `json:"current_owner_rewards"      bson:"current_owner_rewards"`
```
Comment (Woynert) [inline api/interfaces/interfaces.go:77]: The owner cannot get his rewards, the returning array is null.
image

To fix it, the property `current_owner_rewards` needs to be renamed to `current_owners_rewards`.

**Review by Woynert — CHANGES_REQUESTED** [review]

File: api/interfaces/interfaces.go (hunk 2 above)
Comment (PedroChaparro) [reply api/interfaces/interfaces.go]: Done

File: algorithms/database/models/mongoose.js (hunk 1 above)
Comment (PedroChaparro) [reply algorithms/database/models/mongoose.js:45]: Thanks!

**Review by Woynert — APPROVED** [review]
Very good

**Pull request merged**